### PR TITLE
[Infra] Distribution with CD

### PR DESCRIPTION
**Pull request by Lewiky**
This PR sets up SBT and Travis for performing release ready builds on CD pushes to master.

Using the [sbt-assembly](https://github.com/sbt/sbt-assembly) plugin, a MonoJar is built which is executable with `java -jar temple-latest.jar`.

Travis is setup to run this task as a part of it's deploy stage, which then gets pushed to a release.

### CODE REVIEW

File: build.sbt [hunk 1]
```diff
@@ -32,3 +35,12 @@ coverageExcludedPackages := "<empty>;temple\\.Main;"
 
 // Enable formatting on integration tests
 inConfig(IntegrationTest)(org.scalafmt.sbt.ScalafmtPlugin.scalafmtConfigSettings)
+
+//https://stackoverflow.com/questions/28459333/how-to-build-an-uber-jar-fat-jar-using-sbt-within-intellij-idea
```
Comment (jaylees14) [inline build.sbt]: Stick a space between the `//` and the URL 😉

**Review by jaylees14 — APPROVED** [review]
LGTM - how did you test it?

**Comment by c-harding** [comment]
Has this been tested? If so, will accept

**Comment by Lewiky** [comment]
@xsanda @jaylees14 , in response to testing this, CD is now set up to run on any tag being added to the repo - I added a 'test' tag, which can be seen at https://github.com/TempleEight/temple/releases/tag/test-deploy 

Can verify this all works as expected (though there are some issues with Safari not liking downloading files without extensions)

**Review by willjvsmith — APPROVED** [review]
LGTM 🚀

**Review by c-harding — APPROVED** [review]
Looks nice

File: .travis.yml [hunk 2]
```diff
@@ -24,6 +24,22 @@ jobs:
...
+      on:
+        tags: true
+      script: skip
+      before_deploy:
+        - sbt clean assembly
+      deploy:
+        provider: releases
+        skip_cleanup: true
+        api_key:
+          secure: Nuxm6jz9Hqy9jbzoMNBMMBLE+eWNUOu5dAcOuEuWorc76IvKh5Ae2Mwm+8RmOwlH0n5afZg6RZA7HTdv0rALvm/hc611WRksa22rDZsDFThTTuBhEzMTpOWxRG66jzaQAPl8I7RVCJVFidMt28GQuCfCNUbo919CMZmJ6MINGPWUn/NV2FJJ6kLQ7AvLstUqHmhb2143oktg7rtIqgEatIeJ7Yg3jo1HO18i8t0YXZSI7//KIpz9SmXtTqUq5Isi3cGE1eJiu6zhlwQHjirwo425S3QgB64ynCXoHWDJ0sjJezdRcPMA/j7d/SkuRUUq8zbEGsqqUxAbLKn7zd1J3yQZaXGaK54SswYpdTvJXH03cQ43E/Fywf7f+ORzPBBAt6Q7jfZe3IVlS64h0107P2a6+J3X7HW/ik5tFvlIwFV5VzqJMU39J3njQb1Eoe9yHLPWX6kY/L+b8aNZBAvdzxDwepgEgOpWCzbErbwZxZbgmcnw2uNtD4LxfiD48dIqEyyscWv1g4v60LuEKJD5zmw9BILn7O/Ayma8m5prlgz5JpmeXhmI4ZNMZVpCEYeSPS1q7/ENYz+MeOpuSiT8ExrxMJdzFgjY5bkpqfrY5AviAbnL/pujFIULvGq0DC5K8Vf8EPIq6mDFJDSp+5tF/qNR0RFLAHQ7HWjUoMzevN8=
```
Comment (jaylees14) [inline .travis.yml]: why

File: build.sbt [hunk 3]
```diff
@@ -42,3 +49,12 @@ coverageExcludedPackages := "<empty>;temple\\.Main;temple\\.Application"
 
 // Enable formatting on integration tests
 inConfig(IntegrationTest)(org.scalafmt.sbt.ScalafmtPlugin.scalafmtConfigSettings)
+
+// https://stackoverflow.com/questions/28459333/how-to-build-an-uber-jar-fat-jar-using-sbt-within-intellij-idea
+// META-INF discarding
+assemblyMergeStrategy in assembly ~= { _ =>
```
Comment (jaylees14) [inline build.sbt:55]: What is `~=`

**Review by jaylees14 — COMMENTED** [review]
Still not sure we should be committing an API key

**Pull request merged**